pkg/transformers/kubernetes/generate.go

```diff
@@ -0,0 +1,453 @@
+			}
+
+		}
+
+		objects = append(objects, &rc)
+	}
+
+	if len(errs) > 0 {
+		return nil, utilerrs.NewAggregate(errs)
+	}
+
+	// create services for each object with a name based on alias.
+	containers := make(map[string]*kapi.Container)
+	var services []*kapi.Service
+	for _, obj := range objects {
+		switch t := obj.(type) {
+		case *kapi.ReplicationController:
+			ports := app.UniqueContainerToServicePorts(app.AllContainerPorts(t.Spec.Template.Spec.Containers...))
+			if len(ports) == 0 {
+				continue
+			}
+			svc := app.GenerateService(t.ObjectMeta, t.Spec.Selector)
+			if aliases[svc.Name].Len() == 1 {
+				svc.Name = aliases[svc.Name].List()[0]
+			}
+			svc.Spec.Ports = ports
+			services = append(services, svc)
+
+			// take a reference to each container
+			for i := range t.Spec.Template.Spec.Containers {
+				c := &t.Spec.Template.Spec.Containers[i]
+				containers[c.Name] = c
+			}
+		}
+	}
+	for _, svc := range services {
+		objects = append(objects, svc)
+	}
+
+	// for each container that defines VolumesFrom, copy equivalent mounts.
+	// TODO: ensure mount names are unique?
+	for target, otherContainers := range volumesFrom {
+		for _, from := range otherContainers {
+			for _, volume := range containers[from].VolumeMounts {
+				containers[target].VolumeMounts = append(containers[target].VolumeMounts, volume)
+			}
+		}
+	}
+
+	// add emptyDir volume for every volume defined in container
+	for _, obj := range objects {
+		switch o := obj.(type) {
+		case *kapi.ReplicationController:
+			for _, container := range o.Spec.Template.Spec.Containers {
+				for _, volumeMount := range container.VolumeMounts {
+
+					o.Spec.Template.Spec.Volumes = append(o.Spec.Template.Spec.Volumes,
+						kapi.Volume{
+							Name:         volumeMount.Name,
+							VolumeSource: kapi.VolumeSource{EmptyDir: &kapi.EmptyDirVolumeSource{}},
+						})
+				}
+			}
+		}
+	}
+
+	list.Items = objects
+
+	return list, nil
+}
+
+// extractFirstPorts converts a Docker compose port spec (CONTAINER, HOST:CONTAINER, or
+// IP:HOST:CONTAINER) to the first container and host port in the range.  Host port will
+// default to container port.
+func extractFirstPorts(port string) (container, host string) {
+	segments := strings.Split(port, ":")
+	container = segments[len(segments)-1]
+	container = rangeToPort(container)
+	switch {
+	case len(segments) == 3:
+		host = rangeToPort(segments[1])
+	case len(segments) == 2 && net.ParseIP(segments[0]) == nil:
+		host = rangeToPort(segments[0])
+	default:
+		host = container
+	}
+	return container, host
+}
+
+func rangeToPort(s string) string {
+	parts := strings.SplitN(s, "-", 2)
+	return parts[0]
+}
+
+// warnUnusableComposeElements add warnings for unsupported elements in the provided service config
+func warnUnusableComposeElements(k string, v *project.ServiceConfig, warnings map[string][]string) {
+	fn := func(msg string) {
```

Comment:
Right now if there is build it erros out during conversion to kubernetes. see: https://github.com/kadel/henge/blob/add-kubernetes/pkg/transformers/kubernetes/generate.go#L63,L64

My thought was that in future Henge might be able to build image and push it to user provided registry.